src/test/java/bio/terra/service/snapshot/SnapshotConnectedTest.java

```diff
@@ -635,6 +635,98 @@ public void testExcludeLockedFromSnapshotFileLookups() throws Exception {
+            "/mm/" + Names.randomizeName("testdir") + "/testExcludeLockedFromSnapshotFileLookups.txt";
+        FileLoadModel fileLoadModel = new FileLoadModel()
+            .sourcePath(sourceUri.toString())
+            .description("testExcludeLockedFromSnapshotFileLookups")
+            .mimeType("text/plain")
+            .targetPath(targetFilePath)
+            .profileId(billingProfile.getId());
+        FileModel fileModel = connectedOperations.ingestFileSuccess(datasetRefSummary.getId(), fileLoadModel);
+
+        // generate a JSON file with the fileref
+        String jsonLine = "{\"name\":\"name1\", \"file_ref\":[\"" + fileModel.getFileId() + "\"]}\n";
+
+        // load a JSON file that contains the table rows to load into the test bucket
+        String jsonFileName = "this-array-better-pass.json";
+        String dirInCloud = "scratch/testExcludeLockedFromSnapshotFileLookups/" + UUID.randomUUID().toString();
+        BlobInfo ingestTableBlob = BlobInfo
+            .newBuilder(testConfig.getIngestbucket(), dirInCloud + "/" + jsonFileName)
+            .build();
+        Storage storage = StorageOptions.getDefaultInstance().getService();
+        storage.create(ingestTableBlob, jsonLine.getBytes(StandardCharsets.UTF_8));
+
+        // make sure the JSON file gets cleaned up on test teardown
+        connectedOperations.addScratchFile(dirInCloud + "/" + jsonFileName);
+
+        // ingest the tabular data from the JSON file we just generated
+        String gsPath = "gs://" + testConfig.getIngestbucket() + "/" + dirInCloud + "/" + jsonFileName;
+        IngestRequestModel ingestRequest1 = new IngestRequestModel()
+            .format(IngestRequestModel.FormatEnum.JSON)
+            .table("tableA")
+            .path(gsPath);
+        connectedOperations.ingestTableSuccess(datasetRefSummary.getId(), ingestRequest1);
+
+        // create a snapshot
+        SnapshotSummaryModel snapshotSummary = connectedOperations.createSnapshot(
+            datasetRefSummary, "simple-with-filerefs-snapshot.json", "");
+
+        // check that the snapshot metadata row is unlocked
+        String exclusiveLock = snapshotDao.getExclusiveLockState(UUID.fromString(snapshotSummary.getId()));
+        assertNull("snapshot row is unlocked", exclusiveLock);
+
+        String fileUri = getFileRefIdFromSnapshot(snapshotSummary);
+        DrsId drsId = drsIdService.fromUri(fileUri);
+        DRSObject drsObject = connectedOperations.drsGetObjectSuccess(drsId.toDrsObjectId(), false);
+        String filePath = drsObject.getAliases().get(0);
+
+        // lookup the snapshot file by DRS id, make sure it's returned (lookupSnapshotFileSuccess will already check)
+        FileModel fsObjById =
+            connectedOperations.lookupSnapshotFileSuccess(snapshotSummary.getId(), drsId.getFsObjectId());
+        assertEquals("Retrieve snapshot file by DRS id matches desc", fsObjById.getDescription(),
+            fileLoadModel.getDescription());
```

Comment:
It seems kind of brittle to rely on the file description here, but if there are no alternatives you could include a uuid in the description so that you know it's from this test.

Reply:
I do not remember if we had this discussion when we chose the desc for testExcludeLockedFromSnapshotFileLookups which this test steals heavily from. The uuid sounds great to me
(do I want to update the other test that uses this?)

Reply:
As you mentioned in stand-up, I think it's fine to change the other tests in a follow-up PR!